sound/soc/sof/ipc.c

```diff
@@ -279,10 +283,7 @@ int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 	ipc_log_header(sdev->dev, "ipc tx", msg->header);
 
 	/* now wait for completion */
-	if (!ret)
-		ret = tx_wait_done(ipc, msg, reply_data);
-
-	return ret;
+	return tx_wait_done(ipc, msg, reply_data);
```

Comment:
@lyakh I dont relate the logic and comment below for snd_sof_dsp_send_msg()here. From the comment below, it looks like we should call tx_wait_done() only when ret == 0 and not call it for ret > 0. But here we call it for all cases no?

Reply:
@ranj063 I need to add a comment there: if `ret > 0` `msg->ipc_complete` will remain `true` and `tx_wait_done()` will return immediately.